django/thunderstore/api/cyberstorm/views/markdown.py

```diff
@@ -0,0 +1,77 @@
+from typing import Optional
+
+from django.http import Http404
+from rest_framework import serializers
+from rest_framework.generics import RetrieveAPIView, get_object_or_404
+
+from thunderstore.api.utils import CyberstormAutoSchemaMixin
+from thunderstore.markdown.templatetags.markdownify import render_markdown
+from thunderstore.repository.models import Package, PackageVersion
+
+
+class CyberstormMarkdownResponseSerializer(serializers.Serializer):
+    html = serializers.CharField()
+
+
+class PackageVersionReadmeAPIView(CyberstormAutoSchemaMixin, RetrieveAPIView):
+    """
+    Return README.md prerendered as HTML.
+
+    If no version number is provided, the latest version is used.
+    """
+
+    serializer_class = CyberstormMarkdownResponseSerializer
+
+    def get_object(self):
+        package_version = get_package_version(
+            self.kwargs["namespace_id"],
+            self.kwargs["package_name"],
+            self.kwargs.get("version_number"),
+        )
+
+        return {"html": render_markdown(package_version.readme)}
+
+
+class PackageVersionChangelogAPIView(CyberstormAutoSchemaMixin, RetrieveAPIView):
+    """
+    Return CHANGELOG.md prerendered as HTML.
+
+    If no version number is provided, the latest version is used.
+    """
+
+    serializer_class = CyberstormMarkdownResponseSerializer
+
+    def get_object(self):
+        package_version = get_package_version(
+            self.kwargs["namespace_id"],
+            self.kwargs["package_name"],
+            self.kwargs.get("version_number"),
+        )
+
+        if package_version.changelog:
+            return {"html": render_markdown(package_version.changelog)}
+
+        return {"html": ""}
```

Comment:
Use `None` instead of an empty string, this is an example scenario where it does make a difference (lack of content vs. content but empty).

Alternatively returning 404 would be entirely valid here, possibly even preferred as far as REST design practices go

Reply:
@MythicManiac do you foresee this endpoint getting more usage than it currently has in the near future? I'm asking because returning 404 would be ok with current usage (client shouldn't query this endpoint unless there's a changelog). But if the endpoint is reused, returning None would be more robust since Dapper doesn't currenly handle 404s (unhandled error redirects user to 500 page) and I'd rather not scope-creep this comment to implement all that.

Reply:
Can't immediately think of other uses for this endpoint at least.

The reason I'd prefer using status codes is because it's more useful on the load balancing / networking / caching level (since those are actually readable there whereas request bodies often not), and I'd also imagine it's a pattern we might need to use in the future too (e.g. 202 could indicate that an async task is still in progress but currently has no return value)

Not the biggest deal if this isn't addressed right now, but I don't think it's a bad thing to have some pressure on implementing status code handling

Reply:
Handled in https://github.com/thunderstore-io/Thunderstore/pull/969